core/paychannel_manager/impl/paychannel_manager_impl.cpp

```diff
@@ -0,0 +1,261 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "paychannel_manager/impl/paychannel_manager_impl.hpp"
+#include "common/todo_error.hpp"
+#include "paychannel_manager/impl/paychannel_manager_error.hpp"
+#include "vm/actor/builtin/init/init_actor.hpp"
+#include "vm/actor/builtin/payment_channel/payment_channel_actor.hpp"
+#include "vm/message/message_util.hpp"
+#include "vm/state/impl/state_tree_impl.hpp"
+
+namespace fc::paychannel_manager {
+  using fc::vm::message::cid;
+  using vm::VMExitCode;
+  using vm::actor::kInitAddress;
+  using vm::actor::kPaymentChannelCodeCid;
+  using vm::actor::MethodParams;
+  using vm::actor::builtin::payment_channel::State;
+  using vm::message::kDefaultGasLimit;
+  using vm::message::kDefaultGasPrice;
+  using vm::message::kMessageVersion;
+  using vm::message::UnsignedMessage;
+  using vm::state::StateTreeImpl;
+  using InitActorExec = vm::actor::builtin::init::Exec;
+  using PaymentChannelConstruct =
+      vm::actor::builtin::payment_channel::Construct;
+
+  PayChannelManagerImpl::PayChannelManagerImpl(std::shared_ptr<Api> api,
+                                               std::shared_ptr<Ipld> ipld)
+      : api_{std::move(api)}, ipld_{std::move(ipld)} {}
+
+  outcome::result<std::tuple<Address, CID>>
```

Comment:
Address must be optional
```suggestion
  outcome::result<std::tuple<boost::optional<Address>, CID>>
```

Maybe use pair
```suggestion
  outcome::result<std::pair<boost::optional<Address>, CID>>
```

Reply:
Returned type changed, please check.